### ISSUE TITLE
Older siblings don't count for gene score.

### ISSUE THREAD

**Issue by Tarr-OHOL** [jasonrohrer/OneLife#833]
This is intentional and because of gene score and leadership is flawed. 

You purposely did this which when bad actors aren't present is good. You don't want to earn score off older people who aren't doing stuff. Here's where the issue falls - When you're younger and decide to murder your town with others you don't lose points for killing anyone older than yourself. Now lets add in being a younger female. The only people who you murder that lower your score are Grandma - Mom - Younger siblings - Your kids. 
This means every man around you can be killed freely. 
This means all your aunts can be killed freely. 

Gene score and leadership right now are being heavily abused to murder towns and because the trolls are basically losing no points for doing it. I know I say this all the time but I absolutely loathe leadership and posse because of this stuff. With trolls working together what is occurring is one bad actor is exiling someone and having the other trolls attack people. You can only attack the account going bloody but at this point the bad guys have so many alts shared between each other that it's reaching a breaking point. This shit is abysmal to play and as a player I have almost zero agency to actually do anything. 

Gene score needs to probably either tie everyone together and anyone who dies causes point loss or some other fix. Probably also need to undo the short lives changes to actually allow people to posse because most people sids. I just really don't know what to say, you could unfollow your leader but getting 18 people (because families are huge) to unfollow someone is absolutely awful. 

You then need to follow someone else who has to unfollow their troll leader. I had so much damn trouble unfollowing someone because I hadn't realized they were following the troll leader. This shit is just a mess.

**Comment by xMED0x** [comment]
Thanks for bringing up this issue and offering your perspective. I agree that the current gene score system appears to be flawed and in some instances, heavily exploited. As someone who admittedly manipulates the gene score system, I believe my perspective could be valuable in this discussion.

Many players have observed that the gene score system does not reward experienced players in the ways it should. It's not uncommon to see a player who has dedicated an entire lifetime to efficient and productive work in the game, only to open their family tree and discover the tragic fate of their relatives, like some of the following:

Joya, Mother, 25 years old, Killed by a wolf, Last words: "Maybe i should throw myself to a wolf because the dog isn't stopping"
Dina, Little sister, 25 years old, Killed by Wild Boar with Piglet, Last words: "Yikes"
Sina, Little sister, 19 years old, Starved, Last words: "Mom"

Tying gene score to everyone in close proximity might mitigate some of the issues with griefers exploiting the gene score system, but it could introduce new problems. It would inadvertently cause a player's gene score to be heavily impacted by the actions of those around them, which is not necessarily indicative of their individual contributions or experience. Consequently, the gene score might remain unrewarding for genuinely experienced and hardworking players.

One potential solution could be integrating the curse system into the gene score system. I understand this might be challenging to implement, but it could bring about some improvements. Griefers and inexperienced leaders who irresponsibly grant leadership to griefers would be cursed and, as a result, their gene scores would be affected. This would serve as a deterrent and prevent such individuals from repeating their destructive actions.

Moreover, tying the curse system to the gene score would also reward experienced and helpful players who do not get cursed as often. They would end up with higher gene scores, which would make it more rewarding to be a 'good' player in the game. This approach could potentially rectify some of the issues that the community is currently facing with the gene score system.

The idea does bear resemblance to the honor system in League of Legends. In that context, the system rewards players based on positive feedback from other participants, promoting good behavior and cooperation within the game. Similarly, integrating the curse system into the gene score calculation in our game would incentivize positive interactions and discourage harmful behaviors.

In this proposed model, players who behave constructively and contribute positively to the game could receive better gene scores due to fewer curses, simularly to earning honor in League of Legends. On the other hand, disruptive players would accumulate curses, which would impact their gene scores negatively, deterring them from continuing their damaging actions.

**Comment by selb** [comment]
A lifetime of efficient and productive work by Medo, one of 7 this month with 3+ kills:

```
$ grep '6338408' 2023_06_10
B 1686433424 6338408 f35461acc368188c627664bd321105c2ea0a20c7 M (-411438,-72) parent=6338339 pop=76 chain=41
D 1686435230 6338430 319769e4cafa13de1d27f912cab2d7d6d8e96be8 age=25.13 F (-411437,-16) killer_6338408 pop=69
D 1686435403 6338402 ab3183e62987e864bfc1145dd4fe71acd56be3ba age=35.00 F (-411438,-75) killer_6338408 pop=67
D 1686435666 6338493 ab3183e62987e864bfc1145dd4fe71acd56be3ba age=3.86 F (-411443,-74) killer_6338408 pop=62
D 1686436397 6338520 18f5b5f9a35b96b280e1d5f7e9d712561ec8de38 age=5.21 F (-411437,-73) killer_6338408 pop=57
D 1686436649 6338503 e75d8bf90767dd8583fdb0e889a3f4e3e54a6956 age=14.87 F (-411440,-86) killer_6338408 pop=57
D 1686437024 6338408 f35461acc368188c627664bd321105c2ea0a20c7 age=60.00 M (-411541,-286) oldAge pop=60
```

**Comment by xMED0x** [comment]
Ah, selb, always the detective, aren't you? You've managed to catch me in the act of... being productive? I'll take it as a compliment. Also I must point out that between the second and third kill, I had an epiphany about the gene score. It was quite the meditative experience. 

All jokes aside, despite the murders I committed, I noticed something peculiar about the gene score. It didn't seem to hinder me in any way. In fact, it almost felt like I wasn't penalized at all. I'd have thought that committing such disruptive acts would've had a significant impact on my gene score, but it didn't. That is the fundamental flaw we're discussing here, isn't it? Isn't it interesting, and somewhat concerning, that the current gene score system allows for this kind of behavior without adequately punishing it?

Perhaps this is an instance where someone like me, who has admittedly exploited the system, can offer some insight into how it might be improved. I still stand by my earlier suggestion that integrating the curse system into the gene score could help address these issues. Until then, we're left with a system that doesn't necessarily reflect the true consequences of our actions. Just some food for thought.

**Comment by Tarr-OHOL** [comment]
Cursing and gene score do not need to be mixed up. The curse system is tied to allowing the players to self moderate by having a means to block players from coming back after being removed. Before curses we had to have lineage/area bans to prevent the same person from being constantly reborn. 

Gene score is meant to be a score that shows how well a player is able to help their families thrive and go. The problem of course is you actively using your high score to murder and troll which isn't negatively impacting your score because of how the current system works. In theory if you're sitting around enabling the murder of your family either through bears or actual murder you would lose points. However, because of how the game counts family members you're getting free reign to troll.

The fix is unfortunately going to be something along the lines of everyone in your family counting towards score. In the current iteration of the game it's impossible for pseudo eves to exist so in 90% of cases a family is going to live in one town anyways. You might not be interacting with everyone all the time but most of your family is going to be effected by what you're doing good or bad anyways. 

The problem with everyone counting is the same issue as just setting only players in your area counting score would have. 

1). If players only count towards your score in a certain range the griefing method is going to be using a leader out of the towns point loss range denoting a second account as the troll king. Troll king will then walk to town and start being chaotic and will take the point loss for griefing. This isn't going to matter to the troll because losing points won't impact their original accounts high gene score for getting leader which is obviously bad. Players WILL abuse having the ability to ignore score losses by proxy trolling. 

2). If everyone counts for your score your hidden sister/aunt/cousin is going to nuke everyone else's scores by hiding somewhere within a band to kill all their children. Because in the current version of the game you cannot curse your bad mother players are going to respawn to the baby killing troll which then ruins a whole towns gene score. However, maybe an account can only count against you once per life? This means someone constantly killing babies is only hitting you once per player left to starve/murdered. With all the players in town counting towards your score one gene score terrorist should be mostly counteracted. You could also very easily just reintroduce the bad mom code to prevent this sort of trolling. 

Overall, making everyone count is going to be a much more violative solution but actually means people who care about their score need to kill bears or stop trolls instead of sit around and bake pies. Doing an area based solution will lead to trolls just sending disposable pawns out to exile and cause chaos.

**Comment by cordy-123** [comment]
I think that the current gene score system is FINE. 

NO need to change it.

All the worst griefers have mid to bad scores anyway. 

GENE SCORE doesn't matter to the worst ohol griefers. Coz they have bad scores. 

They get leadership thru SCAMMING the current leader to get them to follow them. 

For the most part, **THEY DO _NOT_ GET LEADERSHIP THRU GENE SCORE.**

If griefers can't kill by being a leader (exile and murder), they can do the easier way of bears.

Plus, we do have the Bloody Weapon Vulnerability.

-----

The current gene score system works most of the time. 

Best to leave it.

**Comment by xMED0x** [comment]
lol

**Comment by jasonrohrer** [comment]
Yeah, I will change it to be anyone related to you who dies after you're old enough to help them.

Yeah, there will be more variance in gene score as a result, but variance is just part of it.

The idea is that the best and most family conscious players will rise to the top eventually.

Yes, there will be situation where your uncle is out of town and dies, even though you never met him.  But those random situations will effect everyone equally.

**Comment by jasonrohrer** [comment]
Finally, no matter what, there will sometimes be troll kings.  Heck, we have those in real life all the time.

They are an interesting source of drama in the game, so I'm not seeking to eliminate them completely.

"A mess" is the goal, in terms of the complicated/challenging/weird/unfortunate political situations that you find yourself in.  "In this one life, my leader was a tyrant, so I picked a new person to follow, but then later I realized that the person I was following was secretly still following the tyrant."  That is an emergent story.
